provider/resources.go

```diff
@@ -989,12 +989,31 @@ func Provider() tfbridge.ProviderInfo {
 					Source: "compute_backend_service_signed_url_key.html.markdown",
 				},
 			},
-			"google_compute_disk":                          {Tok: gcpResource(gcpCompute, "Disk")},
-			"google_compute_firewall":                      {Tok: gcpResource(gcpCompute, "Firewall")},
-			"google_compute_firewall_policy":               {Tok: gcpResource(gcpCompute, "FirewallPolicy")},
-			"google_compute_firewall_policy_rule":          {Tok: gcpResource(gcpCompute, "FirewallPolicyRule")},
-			"google_compute_firewall_policy_association":   {Tok: gcpResource(gcpCompute, "FirewallPolicyAssociation")},
-			"google_compute_forwarding_rule":               {Tok: gcpResource(gcpCompute, "ForwardingRule")},
+			"google_compute_disk":                        {Tok: gcpResource(gcpCompute, "Disk")},
+			"google_compute_firewall":                    {Tok: gcpResource(gcpCompute, "Firewall")},
+			"google_compute_firewall_policy":             {Tok: gcpResource(gcpCompute, "FirewallPolicy")},
+			"google_compute_firewall_policy_rule":        {Tok: gcpResource(gcpCompute, "FirewallPolicyRule")},
+			"google_compute_firewall_policy_association": {Tok: gcpResource(gcpCompute, "FirewallPolicyAssociation")},
+			"google_compute_forwarding_rule": {
+				Tok: gcpResource(gcpCompute, "ForwardingRule"),
+				TransformFromState: func(_ context.Context, pMap resource.PropertyMap) (resource.PropertyMap, error) {
+					// The service_directory_registrations field changes from [] in v6 to an object in v7.
+					// It appears that on early versions of v7, upstream has state migrators; they are lost by v7.38
```

Comment:
Can you please link these state migrators? I have not been able to find any.

Is this not a MaxItemsOne change? If so, this should have been caught by the type checker in the bridge, as the type of the property has changed. This might be an effect of PRC, I'll try to test this in https://github.com/pulumi/pulumi-terraform-bridge/issues/2418

Reply:
Actually, this probably never worked for the type checker - it only checks the inputs, not the state.

Reply:
This is likely https://github.com/pulumi/pulumi-terraform-bridge/issues/1667 rather - potentially made worse by PRC as we now save and expect the empty list in state for block types.

Reply:
The only evidence I have here is what I said in the original bug issue - the upgrade works v6 -> v7 -> v8 but not v6 -> v8. It's entirely possible that our rollout of continued support for maxItemsOne is responsible somehow - v6 is two major versions back at this point.